### PR TITLE
8326763: Consolidate print methods in ContiguousSpace

### PR DESCRIPTION
**Pull request by albertnetymk**
Trivial inlining and removing some unused methods.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8326763](https://bugs.openjdk.org/browse/JDK-8326763): Consolidate print methods in ContiguousSpace (**Enhancement** - P4)

### Reviewers
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**)
 * [Guoxiong Li](https://openjdk.org/census#gli) (@lgxbslgx - Committer) ⚠️ Review applies to [5f8f543e](https://git.openjdk.org/jdk/pull/18022/files/5f8f543e9d427b01fa4d30fdc45660a229b633ec)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18022/head:pull/18022` \
`$ git checkout pull/18022`

Update a local copy of the PR: \
`$ git checkout pull/18022` \
`$ git pull https://git.openjdk.org/jdk.git pull/18022/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18022`

View PR using the GUI difftool: \
`$ git pr show -t 18022`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18022.diff">https://git.openjdk.org/jdk/pull/18022.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18022#issuecomment-1966172769)

### CODE REVIEW

File: src/hotspot/share/gc/shared/space.cpp [hunk 1]
```diff
@@ -99,18 +99,9 @@ void ContiguousSpace::mangle_unused_area_complete() {
...
 
-
-void ContiguousSpace::print_short() const { print_short_on(tty); }
-
-void ContiguousSpace::print_short_on(outputStream* st) const {
-  st->print(" space " SIZE_FORMAT "K, %3d%% used", capacity() / K,
-              (int) ((double) used() * 100 / capacity()));
-}
-
-void ContiguousSpace::print() const { print_on(tty); }
```
Comment (stefank) [inline src/hotspot/share/gc/shared/space.cpp:110]: It is a common pattern in HotSpot to have a `print()` function that delegates to `print_on(tty)`. I think this function should be kept.

**Review by stefank — CHANGES_REQUESTED** [review]

**Review by lgxbslgx — APPROVED** [review]
Looks good.

**Review by stefank — APPROVED** [review]

File: src/hotspot/share/gc/shared/space.cpp [hunk 2]
```diff
@@ -99,20 +99,12 @@ void ContiguousSpace::mangle_unused_area_complete() {
...
-              (int) ((double) used() * 100 / capacity()));
-}
-
 void ContiguousSpace::print() const { print_on(tty); }
 
 void ContiguousSpace::print_on(outputStream* st) const {
-  print_short_on(st);
-  st->print_cr(" [" PTR_FORMAT ", " PTR_FORMAT ", " PTR_FORMAT ")",
-                p2i(bottom()), p2i(top()), p2i(end()));
+  st->print_cr(" space %zuK, %3d%% used [" PTR_FORMAT ", " PTR_FORMAT ", " PTR_FORMAT ")",
```
[the remark this answers is not in the document]
Comment (albertnetymk) [reply src/hotspot/share/gc/shared/space.cpp:105]: `%zu` is shorter and easier to read, and it's widely used in new code; `SIZE_FORMAT` is used in old code when we didn't have `%zu`.

[the remark this answers is not in the document]
Comment (lgxbslgx) [reply src/hotspot/share/gc/shared/space.cpp:105]: Got it. Thanks for your explanation.

**Comment by albertnetymk** [comment]
Thanks for review.

/integrate

**Pull request closed without merging**